### PR TITLE
Fixes #6446 - Forbidding non-json POST/PUT requests in v2

### PR DESCRIPTION
**Pull request by daviddavis**
## Tasks
- [x] Fix any broken tests
- [x] Write new tests

### CODE REVIEW

**Comment by isratrade** [comment]
@daviddavis, you need to add this to test_helper.rb with the `Spork.each_run` block

``` ruby
      @request.env['CONTENT_TYPE'] = 'application/json'
      @request.env['HTTP_ACCEPT'] = 'application/json'
```

I think it's fine to put inside the `set_basic_auth` method

**Comment by daviddavis** [comment]
@isratrade sounds good. thanks.

**Comment by daviddavis** [comment]
This should be ready for review.

File: app/views/api/v2/errors/unsupported_content_type.json.rabl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+object false
+
+node(:message) { "Content type '#{request.content_type}' unsupported in v2 for POST and PUT requests. Please use 'application/json'." }
```
Comment (isratrade) [inline app/views/api/v2/errors/unsupported_content_type.json.rabl]: @daviddavis, nitpick, add "is" before "unsupported" and some other small edits

``` ruby
node(:message) { "Content-Type:#{request.content_type} is unsupported in API v2 for POST and PUT requests. Please use 'Content-Type:application/json'." }
```

**Comment by daviddavis** [comment]
@isratrade updated the message. thanks.

File: app/views/api/v2/errors/unsupported_content_type.json.rabl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+object false
+
+node(:message) { "'Content-Type: #{request.content_type}' is unsupported in API v2 for POST and PUT requests. Please use 'Content-Type: application/json'." }
```
Comment (domcleal) [inline app/views/api/v2/errors/unsupported_content_type.json.rabl]: Please extract this, i.e. `_("'Content-Type: %s' is unsup....") % request.content_type`

**Comment by domcleal** [comment]
NItpick, ideally the test would be in test/functional/api/base_controller_subclass_test.rb.

Otherwise good.

File: app/views/api/v2/errors/unsupported_content_type.json.rabl (hunk 2 above)
Comment (daviddavis) [inline app/views/api/v2/errors/unsupported_content_type.json.rabl]: Can do. I did notice though that none of the other templates aren't i18nized.

Comment (domcleal) [inline app/views/api/v2/errors/unsupported_content_type.json.rabl]: Ugh, I'll file a bug for that, thanks for spotting.  Would you mind doing this one?

http://projects.theforeman.org/issues/6864

Comment (daviddavis) [inline app/views/api/v2/errors/unsupported_content_type.json.rabl]: Definitely. Thanks.

**Comment by domcleal** [comment]
http://paste.fedoraproject.org/122282/81655614/ should fix the test failure

**Comment by domcleal** [comment]
Merged as 50ebc02 for Foreman 1.6, thanks @daviddavis!

**Pull request closed without merging**